linode/manager
# upcoming: [M3-8306] - Hide CORS and SSL for OBJ Gen2

## Description 📝
CORS (Cross Origin Sharing) is not available for endpoint types E2 and E3. We will not be supporting SSL/TLS either at the moment.

## Changes  🔄
### For Gen2 (E2 & E3):
- Hide CORS and display notice
- Disable/Hide SSL/TLS Tab

## Target release date 🗓️
N/A

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-08-12 at 10 15 02 PM](https://github.com/user-attachments/assets/ee01afdc-1959-4aca-8075-62f5684dd7ad) | ![Screenshot 2024-08-12 at 10 14 49 PM](https://github.com/user-attachments/assets/c19ada2b-1c6e-4fb6-8a7d-44499b7fde5f) |

## How to test 🧪
>[!Note]
We have E2E tests ticket M3-8440 to cover all these scenarios separately.

### Prerequisites
- Enable MSW and ensure `OBJ Gen2` flag is enabled

### Verification steps
>[!Note]
This will work without refreshing the page only after you've navigated to the bucket details page, as the bucket names are randomly generated each time due to MSW.
- Select a bucket
- When on details page, observe SSL/TLS tab is not rendered
- Go to `Access` tab and observe CORS is not rendered

## As an Author I have considered 🤔

*Check all that apply*
